src/client/common/alias_dict.cpp

```diff
@@ -229,3 +430,50 @@ void mp::AliasDict::save_dict()
             throw std::runtime_error(fmt::format("cannot create aliases config file {}", config_file_name));
     }
 }
+
+// This function removes the contexts which do not contain aliases, except the active context.
+void mp::AliasDict::sanitize_contexts()
+{
+    // To avoid invalidating iterators, the function works in two stages. First, the aliases which need to be
+    // removed are determined and, second, they are effectively removed.
+    std::vector<std::string> empty_contexts;
+
+    for (auto& context : aliases)
+        if (context.first != active_context && context.second.empty())
+            empty_contexts.push_back(context.first);
+
+    if (!empty_contexts.empty())
+    {
+        modified = true;
+        for (const auto& context : empty_contexts)
+            aliases.erase(context);
+    }
+}
+
+// Returns an alias definition iff the given alias name is unique across all the contexts. The given alias name cannot
+// be fully qualified, that is, it must not be prepended by a context name.
+std::optional<mp::AliasDefinition> mp::AliasDict::get_alias_from_all_contexts(const std::string& alias) const
+{
+    const AliasDefinition* ret;
+    bool found{false};
+
+    for (const auto& context : aliases)
+    {
+        const auto& context_dict = context.second;
+
+        auto found_alias = context_dict.find(alias);
+
+        if (found_alias != context_dict.cend())
+        {
+            if (found) // This means that the alias exists in more than one context.
+                return std::nullopt;
```

Comment:
Since the `else` has braces, I'd prefer braces for the `if()` as well for readability.

Reply:
Right, done.

Reply:
Done.